### PR TITLE
fix Issue 3281 by merging stores after expression statements

### PR DESCRIPTION
**Pull request by d367wang**
This PR fixes https://github.com/typetools/checker-framework/issues/3281 by merging stores after expression statements, as follows. 

- When building the CFG, the top-level nodes of all the expression statements are saved in a set. 
- In analyzing phase, check whether the visited node is in the above set while containing both then- and else- stores. If true, merge the two stores.

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/ControlFlowGraph.java [hunk 1]
```diff
@@ -124,14 +129,38 @@ public SpecialBlock getEntryBlock() {
...
+     * Returns all return nodes of the control flow graph.
+     *
+     * @return all return nodes of the control flow graph.
+     */
     public List<ReturnNode> getReturnNodes() {
         return returnNodes;
     }
 
+    /**
+     * Return the top-level nodes of all the expression statements encountered.
```
Comment (xingweitian) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/ControlFlowGraph.java:142]: To be consistent, here maybe also use `Returns`?

Comment (d367wang) [reply dataflow/src/main/java/org/checkerframework/dataflow/cfg/ControlFlowGraph.java:142]: Okay, thanks for pointing out!

File: dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java [hunk 2]
```diff
@@ -3616,7 +3669,11 @@ public Node visitErroneous(ErroneousTree tree, Void p) {
 
         @Override
         public Node visitExpressionStatement(ExpressionStatementTree tree, Void p) {
-            return scan(tree.getExpression(), p);
+            Node node = scan(tree.getExpression(), p);
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java:3672]: Thanks for exploring this option.
Can you investigate the following alternative: Here, create a local variable and assign non-void expressions to that local variable.
I think that should also cause merging of the two branches and is far less invasive than adding `expressionStatementRootNodes`, for which I don't really see a good use case.

**Comment by d367wang** [comment]
This PR tries to solve https://github.com/typetools/checker-framework/issues/3281. However, the solution touches the dataflow framework infrastructure by handling special cases in method `callTransferFunction` in ForwardAnalysisImpl.java, which is a bad design.

**Pull request closed without merging**